src/components/SubHeader/SubHeader.js

```diff
@@ -0,0 +1,85 @@
+import "./SubHeader.css";
+import ratingIcon from "../../assets/icons/star.svg";
+import MenuFilter from "./Restaurant/MenuFilter";
+import BreadCrumb from "../../commonComponents/BreadCrumb";
+import getBreadCrumbConfig from "./helpers/breadCrumbConfig";
+import Offers from "./Restaurant/Offers";
+import Feedback from "./Restaurant/Feedback";
+import React from "react";
+import { getRestaurantDetails } from "../../apis/getRestaurantDetails";
+import Header from "./Restaurant/Header";
+import Spinner from "../../commonComponents/Spinner";
+
+class SubHeader extends React.Component {
+  state = {
+    restaurantDetails: {},
+    isLoading: true,
+  };
+
+  componentDidMount() {
+    getRestaurantDetails(1)
+      .then((res) => res.json())
+      .then((data) => {
+        this.setState({
+          restaurantDetails: data,
+          isLoading: false,
+        });
+      });
+  }
+
+  render() {
+    if (this.state.isLoading) {
+      return (
+        <div className="spinner-container">
+          <Spinner text="Loading..." />
+        </div>
+      );
+    }
+
+    const {
+      restaurantDetails: {
+        thumbnail,
+        name,
+        category,
+        address: { street, locality, city },
+        offers,
+        aggregateRating: { ratingCount, ratingValue },
+        deliveryTime,
+        priceRange: { cost, count },
+      },
+    } = this.state;
+
+    const breadCrumbConfig = getBreadCrumbConfig([city, locality, name]);
```

Comment:
1. Rule of thumb:- if a helper function is taking more than 3 argument pass them as a object else pass as a parameter
2. If return value here is array or object use defaultMemoize